src/main/java/com/google/roomies/util/StringConverter.java

```diff
@@ -27,13 +29,44 @@ public static Date stringToDate(String date) throws ParseException {
   }
 
   /**
-  * Converts a string price to a Money instance with a USD currency code.
+  * Converts a string price to a Money instance. 
   *
+  * If currency code is not specified, defaults to USD.
+  * @param moneyRepresentation string representation of a monetary amount
+  * @return a Money instance of the input price.
+  */
+  public static Money stringToMoney(String moneyRepresentation) throws
+     UnknownCurrencyException,MonetaryParseException, NumberFormatException {
+    Money convertedPrice;
+    try {
+      convertedPrice = priceAndCurrencyToMoney(moneyRepresentation);
+    } catch (MonetaryParseException | NumberFormatException e) {
+      convertedPrice = priceToMoney(moneyRepresentation);
+    }
+    return convertedPrice;
+  }
+
+  /**
+  * Converts a string price and currency amount to Money.
+  *
+  * Input should be in the form of "CURRENCY_CODE price" (ex. "USD 28.87").
+  * @param priceAndCurrency string representation of a monetary amount
+  * @return a Money instance of the price in specified currency code
+  */
+  private static Money priceAndCurrencyToMoney(String priceAndCurrency) throws
+    UnknownCurrencyException, MonetaryParseException, NumberFormatException {
+    return Money.parse(priceAndCurrency);
+  }
+
+  /**
+  * Converts a string price amount to Money in USD.
+  *
+  * Input should be in the form of "price" (ex. "28.87").
   * @param price string representation of a monetary amount
-  * @return a Money instance of the price in USD.
+  * @return a Money instance of the price in USD
   */
-  public static Money stringToMoney(String price) {
+  private static Money priceToMoney(String price) {
     CurrencyUnit usd = Monetary.getCurrency(CURRENCY_CODE);
-    return Money.of(Integer.parseInt(price), usd);
+    return Money.of(Integer.parseInt(price.toString()), usd);
```

Comment:
This seems misleading - so "28.87" will fail with a parse exception (it's not an int), and even if that worked you would be silently rounding. 

Also price.toString() is calling toString on a string

Reply:
Changed to convert to a BigDecimal.